extensions/mongodb/pom.xml

```diff
@@ -47,19 +47,82 @@
                     </archive>
                 </configuration>
             </plugin>
+
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <version>${maven.failsafe.plugin.version}</version>
+                <configuration combine.children="append">
+                    <!--
+                    Failsafe uses the jar, but the main artifact, we shade the dependencies into a jar with jar-with-dependencies classifier
+                    Moreover this parameter name is really confusing.
+                    -->
+                    <classesDirectory>${project.build.directory}/${project.artifactId}-${project.version}-jar-with-dependencies.jar</classesDirectory>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
+                <version>${maven.shade.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>jar-with-dependencies</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <filters>
+                                <filter>
+                                    <artifact>*:*</artifact>
+                                    <excludes>
+                                        <exclude>module-info.class</exclude>
+                                        <exclude>META-INF/*.SF</exclude>
+                                        <exclude>META-INF/*.DSA</exclude>
+                                        <exclude>META-INF/*.RSA</exclude>
+                                    </excludes>
+                                </filter>
+                            </filters>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.calcite</pattern>
```

Comment:
you don't relocate other transitive dependencies of hazelcast-sql. If they are used in this module, they will remain unshaded.

Reply:
this is maven, not gradle - there is no "api". If you use guava in `mongodb` module, because you can (as transitive dep of hazelcast-sql) and not relocate the references, it will fail at runtime, because there will be only shaded guava classes.

Reply:
Currently, there is no such use. We could use the dependency plugin to detect use of classes from transitive dependencies

https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html

It would tell us (and possibly failed the build) if we used anything outside of `hazelcast-sql` or `calcite-core` which are declared as direct dependencies. But that comes with other issues (high maintenance, exceptions for reflection etc..), and we should detect any issues in our soak tests.